File: 프로그래머스/Lv2/피보나치 수.py
def solution(n):
    answer = 0
    arr=[0]*100001
    arr[1]=1
    num=2
    
    while(num<=n):
        arr[num]=(arr[num-1]%1234567+arr[num-2]%1234567)%1234567
        num+=1
        
    answer=arr[n]
    return answer

File: 프로그래머스/Lv2/피보나치 수.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    vector <int> v;
    v.push_back(0);
    v.push_back(1);
    for(int i=2;i<=n;i++){
        int temp = (v[i-1]% 1234567 + v[i-2]% 1234567)% 1234567;
        v.push_back(temp);
    }
    answer=v[n];
    return answer;
}
